Update refund creation documentation for fix 27376

Konamiman opened the pull request:
https://github.com/woocommerce/woocommerce/pull/29099 does changes in the refunds creation entry point. This pull request, that should be merged (and the docs re-rendered) together with the former, fixes the docs for that:

- Specifies that `amount` is now optional.
- Adds `refund_total` to line items and taxes.
- Updates refund creation code examples to be more comprehensive.

Konamiman committed:
Updates for the fix 27376 on the WooCommerce core repo.

- Specify that "amount" is now optional.
- Add "refund_amount" to line items and taxes.
- Update refund creation code examples to be more comprehensive.

vedanshujain commented on source/includes/wp-api-v3/_order-refunds.md:
```diff
@@ -4,18 +4,18 @@ The refunds API allows you to create, view, and delete individual refunds.
+| `id`               | integer   | Unique identifier for the resource. <i class="label label-info">read-only</i>                                                                    |
+| `date_created`     | date-time | The date the order refund was created, in the site's timezone. <i class="label label-info">read-only</i>                                         |
+| `date_created_gmt` | date-time | The date the order refund was created, as GMT. <i class="label label-info">read-only</i>                                                         |
+| `amount`           | string    | Total refund amount. Optional on creation, if missing it will be calculated by adding up the `refund_amount` of each line item, including taxes. |
```
We should reword it like: "Total refund amount. Optional. If this parameter is provided, it will take precedence over line item totals, even when total of line items does not matches with this amount." WDYT?

Konamiman replied:
Yep, that sounds better.

vedanshujain commented on source/includes/wp-api-v3/_order-refunds.md:
```diff
@@ -43,14 +43,16 @@ The refunds API allows you to create, view, and delete individual refunds.
+| `id`           | integer | Tax rate ID. <i class="label label-info">read-only</i>                         |
+| `total`        | string  | Tax total. <i class="label label-info">read-only</i>                           |
+| `subtotal`     | string  | Tax subtotal. <i class="label label-info">read-only</i>                        |
+| `refund_total` | number  | The amount to refund for this taxs. <i class="label label-info">write-only</i> |
```
minor typo in `taxs`

vedanshujain reviewed (changes requested):
Great work, along with the main PR. Requesting minor changes.

Konamiman committed:
Better wording for the `amount` attribute.

claudiosanches reviewed (approved):
Looks great!

claudiosanches commented on source/includes/wp-api-v3/_order-refunds.md:
````diff
@@ -70,13 +72,41 @@ curl -X POST https://example.com/wp-json/wc/v3/orders/723/refunds \
 ```javascript
 const data = {
-  amount: "10"
+    "order_refund": {
````
This is an incorrect syntax for JavaScript

Konamiman replied:
Yep, no quotes needed!

claudiosanches commented on source/includes/wp-api-v3/_order-refunds.md:
```diff
@@ -70,13 +72,41 @@ curl -X POST https://example.com/wp-json/wc/v3/orders/723/refunds \
 	-H "Content-Type: application/json" \
 	-d '{
-  "amount": "10"
+  "order_refund": {
```
We only create wraps like that in our legacy API, this is intentional here?

Konamiman replied:
I think I confused request and response, that `"order_refund"` doesn't belong here, indeed. I've fixed it.

claudiosanches reviewed (changes requested):
I found some small issues with this PR.

Konamiman committed:
Fix examples for the order refund creation.

claudiosanches commented on source/includes/wp-api-v3/_order-refunds.md:
```diff
@@ -70,13 +72,35 @@ curl -X POST https://example.com/wp-json/wc/v3/orders/723/refunds \
 	-u consumer_key:consumer_secret \
 	-H "Content-Type: application/json" \
 	-d '{
-  "amount": "10"
```
All other examples includes `amount: 30`, this one suppose to include one too or we should remove from all.

claudiosanches commented on source/includes/wp-api-v3/_order-refunds.md:
```diff
@@ -70,13 +72,35 @@ curl -X POST https://example.com/wp-json/wc/v3/orders/723/refunds \
+  "line_items": [
+    {
+      "id": "111",
+      "refund_amount": 10,
```
Seems like `refund_amount` isn't documented, maybe you mean `refund_total`?

claudiosanches commented on source/includes/wp-api-v3/_order-refunds.md:
```diff
@@ -91,7 +115,19 @@ WooCommerce.post("orders/723/refunds", data)
 <?php
 $data = [
-    'amount' => '10'
+     'amount' => 30,
```
`amount` uses string as documented, need to fix all other occurrences in this file too.

claudiosanches reviewed (changes requested):
@Konamiman Thanks for addressing all issues we found out. I just found a few more that need to be fixed before we can deploy it. Thanks.

Konamiman committed:
Fixes in the documentation for order refunds:

- "refund_amount" fields fixed to "refund_total"
- "amount" values turned into strings
- Added missing "amount" field in one of the examples

Konamiman commented:
@claudiosanches good catches, thanks. I've fixed everything.

claudiosanches reviewed (approved):
Great work!
